PAD-74: Add SBOM scan documentation (#1108)
spectrocloud/librarium: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md

![conformance.png](conformance.png)

# SBOM: Dependencies & Vulnerabilities

## What is an SBOM?

> A Software Bill of Materials (SBOM) is a comprehensive list of the components, libraries, and other assets that make up a software application. It details the various third-party components and dependencies used in the software and helps in managing security and compliance risks associated with those components.

Reviewer: In the last sentence, suggest: 
It details the various third-party components and dependencies used in the software and helps to manage security and compliance risks associated with those components.

> An SBOM provides metadata about each component such as version, origin, license, and more, enabling organizations to track vulnerabilities, perform regular software maintenance, and ensure compliance with regulatory requirements such as the European Union's General Data Protection Regulation (GDPR) and the Payment Card Industry Data Security Standard (PCI DSS).

Reviewer: comma in first sentence before such as can be removed because it introduces a comma list.

Palette will identify every unique container image within your chosen scope and not only generate an SBOM for that image, but also run the SBOM through a vulnerability scanner that will flag CVEs. Palette leverages two open source tools from Anchore for this purpose: [Syft](https://github.com/anchore/syft) for SBOM generation and [Grype](https://github.com/anchore/grype) for vulnerability detection.

If a [backup location](/clusters/cluster-management/backup-restore) is provided, the SBOM for each image will be uploaded to your backup location and can subsequently be downloaded with the click of a button or using the Palette API. If a backup location is not provided, Palette will still preserve all of the dependencies and vulnerabilities that were identified, but the raw SBOMs will not be available for download. All the results and metadata pictured in the screenshots below will be included even without a backup location configured, except when using the **github-json** format, in which case a backup location must be provided.

<br />

<WarningBox>
A Backup Location is <b>mandatory</b> when configuring an SBOM scan using the <b>github-json</b> format.
</WarningBox>

<br />

> #### SBOM scan formats

Reviewer 2: ```suggestion
#### SBOM Scan Format
```

> * [SPDX](https://github.com/spdx/spdx-spec/blob/v2.2/schemas/spdx-schema.json)
>   * A standardized format for representing Software Bill of Materials that is widely used by organizations and governments. It has been around longer than any other SBOM format.
> * [CycloneDX](https://cyclonedx.org/specification/overview/)
>   * An open-source XML format that provides a standard representation of software components and their metadata.
> * GitHub's [dependency submission format](https://docs.github.com/en/rest/dependency-graph/dependency-submission?apiVersion=2022-11-28)
>   * Can be used to power a [dependency review workflow](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-dependency-review).
> * Syft JSON
>   * Syft's custom SBOM format. Contains the most complete metadata out of the four options.

Reviewer 2: ```suggestion
* [SPDX](https://github.com/spdx/spdx-spec/blob/v2.2/schemas/spdx-schema.json): A standardized format for representing SBOM that is widely used by organizations and governments. This report format has been around longer than any other SBOM format.

* [CycloneDX](https://cyclonedx.org/specification/overview/): An open-source XML format that provides a standard representation of software components and their metadata.

* GitHub's [dependency submission format](https://docs.github.com/en/rest/dependency-graph/dependency-submission?apiVersion=2022-11-28): Can be used to power a [dependency review workflow](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-dependency-review).

* Syft JSON:  Syft's custom SBOM format. This format contains the most metadata compared to the other formats.
```
Updated the markdown list to include the descriptions within the same bullet.

> #### SBOM scan scopes

Reviewer 2: ```suggestion
#### SBOM Scan Scopes
```

> * Cluster
>   * Scan **all** images in your K8s cluster
> * Namespace
>   * Scan all images in a particular **Namespace**
> * Label Selector
>   * Scan all images used by all of the Pods matching a **label selector** within a particular Namespace
> * Pod
>   * Scan all images used by a single **Pod**

Reviewer 2: ```suggestion
* Cluster: Scans all the images in your Kubernetes cluster.

* Namespace: Scans all images in a particular Kubernetes namespace.

* Label Selector: Scans all images used by all of the Pods matching a label selector within a particular Kubernetes namespace.

* Pod: Scans all images used by a single Pod.
```

> ## Viewing SBOM scan results

Reviewer 2: ```suggestion
## Review SBOM Results
```

> You can click into a completed scan to view a scan report containing additional detail for every image that was scanned. The context column indicates every unique usage of each image, broken out by container name, namespace, and pod name, as each image may be in use by various containers within a given scope. The vulnerability summary column provides a condensed view of the vulnerability report, which can be viewed in greater detail by clicking on any row in the scan report.

Reviewer 2: ```suggestion
To review a completed scan, expand the **Software Bill of Materials (SBOM)** row. The expanded row displays the completed report containing detailed information about every scanned image. The context column indicates every unique usage of each image, broken out by container name, namespace, and pod name. Each image may be used by various containers within a given scope. The vulnerability summary column provides a condensed view of the vulnerability report, which can be viewed in greater detail by clicking on any row in the scan report.
```

![sbom_results.png](sbom_results.png)

> Each image details page within the scan report provides a list of dependencies and vulnerabilities. These tables are condensed highlights of the metadata contained in the SBOM that was generated for a particular image. Each dependency’s version and type is displayed, but additional metadata will be included in the SBOM. Exactly what additional metadata is included will depend on the selected SBOM format.

Reviewer 2: ```suggestion
Each identified image has its own detailed results page. To review the image report, select the **>** button. Each image's details page within the scan report provides a list of dependencies and vulnerabilities.

The tables contained highlights of the metadata in the SBOM for a particular image. Each dependency's version and type are displayed. The amount of displayed details in the metadata section depends on the selected SBOM format. 
```

Flipped the order and clean up the language.

Reviewer: Suggest:
The details page for each image in the scan report provides a list of dependencies and vulnerabilities.

The tables list highlights of the metadata in the SBOM for a particular image. Each dependency's version and type are displayed. The amount of displayed details in the metadata section depends on the selected SBOM format.

Author: @karl-cardenas-coding @ritawatson one small issue with the re-wording(s) is that the additional metadata is in the SBOM - not on our UI. Users will only notice it if they download the SBOM and inspect it. Our UI shows the same info (dependencies & vulnerabilities) no matter what.

Reviewer 2: @TylerGillson I see, in that case, it sounds like Rita's suggestion is probably closer to what you're thinking. We could end the paragraph by stating the additional metadata is only available in the downloaded format.

![sbom_dependencies.png](sbom_dependencies.png)

> For each vulnerability, you can view the name, severity level, vulnerability code, installed or impacted version, and the fix version (if a fix is available). Any CVEs documented in the [NIST National Vulnerability Database](https://nvd.nist.gov/vuln) (NVD) will render as a hyperlink to the NVD detail page for that particular vulnerability.

Reviewer 2: ```suggestion
For each identified vulnerability, you can view the name, severity level, vulnerability code, installed or impacted version, and the fix version (if a fix is available). Any CVEs documented in the [NIST National Vulnerability Database](https://nvd.nist.gov/vuln) (NVD) will render as a hyperlink to the NVD detail page for that particular vulnerability.
```

![sbom_vulnerabilities.png](/sbom_vulnerabilities.png)

# Scan Options

The following options are available for running cluster scans:
   
## On Demand

> A cluster scan of any type can be started by navigating to the scans tab of a cluster on the UI. The scan shows up as 'initiated' and transitions to 'Completed' when the scan is complete.

Reviewer: ```suggestion
A cluster scan of any type can be started by navigating to the **Scans** tab of a cluster in Palette. Scan progress displays as 'Initiated' and transitions to 'Completed' when the scan is complete.
```

|__On Demand Scan__|
|------------------|
|Select the cluster to scan -> Scan(top panel) -> Run Scan|

## Scheduled

> A schedule can be set for each type of scan at the time of deploying the cluster initially. The schedule can also be (re)set at a later point.

Reviewer: ```suggestion
You can set a schedule for each scan type when you deploy the cluster, and you can change the schedule at a later time.
```

|__During Cluster Deployment__|
|-----------------------------|

> |Add New Cluster -> Cluster Policies -> Scan Policies -> Enable and schedule desired scans|

Reviewer: ```suggestion
|Add New Cluster -> Cluster Policies -> Scan Policies -> Enable and schedule desired scans.|
```

|__Running Cluster__|
|----------------------|

> |Select the cluster to scan -> Settings -> Cluster Settings -> Scan Policies -> Enable and schedule scans of your choice|

Reviewer: ```suggestion
|Select the cluster to scan -> Settings -> Cluster Settings -> Scan Policies -> Enable and schedule scans of your choice.|
```

### Schedule Options Available

> * Schedule your compliance scan for an exact month, day, hour, and minute

Reviewer: ```suggestion
Schedule your compliance scan for month, day, hour, or minute. For example: 
* Every week on Sunday at midnight.
* Every two weeks at midnight.
* Every month on the first day of the month at midnight.
* Every two months on the first day of the month at midnight
```

Reviewer: Made the examples a bullet list and removed the top-level bullet.

It would be useful to include an example of hour and minute to replace one of the week and one of the month examples.

![conformance.png](/conformance.png)

# SBOM: Dependencies & Vulnerabilities

## What is an SBOM?
An SBOM is a comprehensive list of the components, libraries, and other assets that make up a software application. It details the various third-party components and dependencies used in the software and helps to manage security and compliance risks associated with those components.

The SBOM provides metadata about each component such as version, origin, license, and more. Reviewing the SBOM enables organizations to track vulnerabilities, perform regular software maintenance, and ensure compliance with regulatory requirements such as the European Union's General Data Protection Regulation (GDPR) and the Payment Card Industry Data Security Standard (PCI DSS).

![sbom_scan.png](/sbom_scan.png)

## Configure an SBOM Scan
To initiate an SBOM scan, navigate to **Clusters** and select the cluster to scan. On the **Cluster Overview** page, click the **Scans** tab, and expand the **Software Bill of Materials (SBOM)** drop-down menu. Select **Configure Scan** and choose the desired SBOM format, scan scope, and an optional backup location. Confirm your changes.

> Palette will identify every unique container image within your chosen scope and generate an SBOM for that image. Paletee also runs the SBOM through a vulnerability scanner to flag any Common Vulnerabilities and Exposures (CVEs). Palette leverages two open-source tools from Anchore: [Syft](https://github.com/anchore/syft) for SBOM generation and [Grype](https://github.com/anchore/grype) for vulnerability detection.

Reviewer 2: ```suggestion
Palette will identify every unique container image within your chosen scope and generate an SBOM for that image. Palette also runs the SBOM through a vulnerability scanner to flag any Common Vulnerabilities and Exposures (CVEs). Palette leverages two open-source tools from Anchore: [Syft](https://github.com/anchore/syft) for SBOM generation and [Grype](https://github.com/anchore/grype) for vulnerability detection.
```
